Project/archive/index.js

```diff
@@ -0,0 +1,95 @@
+var button = document.querySelector('.anotherjoke');
+let tellJoke = document.querySelector(".telljoke")
+let tellJoke2 = document.querySelector(".telljoke2")
+let isTwoPart = true;
+let onePartText = "default1";
+let twoPartText = "default2";
+
+function myFunction() {
+    document.getElementById("myDropdown").classList.toggle("show");
+}
+
+function displayJoke() {
+  return function(response) {
+    return new Promise(resolve => {
+      if(isTwoPart){tellJoke2.innerHTML = twoPartText}
+      else{tellJoke2.innerHTML = ""}
+    });
+  };
+}
+
+function sleeper(ms) {
+  return function(response) {
+    return new Promise(resolve => setTimeout(() => resolve(response), ms));
+  };
+}
+
+button.addEventListener("click", writeJoke);
+
+writeJoke();
+function writeJoke() {
+  {
+    tellJoke.innerHTML = "Fetching Joke...";
+    tellJoke2.innerHTML = "";
+    fetch('https://v2.jokeapi.dev/joke/Programming').then(response => {
+      if (!response.ok) {
+          throw new Error();
+      }
+      return response.json();
+      }).then(sleeper(4000))
+      .then(data => {
+        if(data['type'] == "single") {
+          isTwoPart = false;
+          tellJoke.innerHTML = data['joke'];
+          tellJoke2.innerHTML =  "";
+          twoPartText = "";
+        }else{
+          isTwoPart = true;
+          tellJoke.innerHTML = data['setup'];
+          twoPartText = data.delivery;
+        }
+    })
+    .then(sleeper(4000))
+    .then(displayJoke())
+    .catch(function (error) {
```

Comment:
```suggestion
    .catch(function () {
```
You don't need to specify error here if you aren't going to use anything from the error being caught.